### PR TITLE
REST implementation of  `/eth/v1/validator/prepare_beacon_proposer`

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

implements following the new model we defined for beacon APIs. 
- [x] /eth/v1/validator/prepare_beacon_proposer
removes the apimiddleware implementation

BenchmarkServer_PrepareBeaconProposer-10    	      63	  17806538 ns/op

**Which issues(s) does this PR fix?**

part of #12635

### CODE REVIEW

**Review by saolyn — COMMENTED** [review]
Could you please add a benchmark.
```
curl [your endpoint] -w "\n%{time_starttransfer}\n"
```

File: beacon-chain/rpc/eth/shared/structs.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ type SignedValidatorRegistration struct {
 	Signature string                 `json:"signature" validate:"required,hexadecimal"`
 }
 
+type FeeRecipient struct {
+	ValidatorIndex string `json:"validator_index"`
+	FeeRecipient   string `json:"fee_recipient" hex:"true"`
```
Comment (rkapka) [inline beacon-chain/rpc/eth/shared/structs.go:91]:
```suggestion
	FeeRecipient   string `json:"fee_recipient"`
```
`validate` tags missing

File: beacon-chain/rpc/eth/validator/handlers.go [hunk 2]
```diff
@@ -571,3 +577,62 @@ func (s *Server) RegisterValidator(w http.ResponseWriter, r *http.Request) {
...
+	}
+	var feeRecipients []common.Address
+	var validatorIndices []primitives.ValidatorIndex
+	// filter for found fee recipients
+	for i, r := range jsonFeeRecipients {
+		validatorIndex, valid := shared.ValidateUint(w, "Validator Index", r.ValidatorIndex)
+		if !valid {
+			return
+		}
+		feeRecipientBytes, err := hexutil.Decode(r.FeeRecipient)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/handlers.go:604]: As discussed in publishBlockV3, we should probably return the hex from `shared.ValidateHex`

File: beacon-chain/rpc/eth/validator/handlers.go [hunk 3]
```diff
@@ -571,3 +577,62 @@ func (s *Server) RegisterValidator(w http.ResponseWriter, r *http.Request) {
...
+		if err != nil {
+			http2.HandleError(w, fmt.Sprintf("fee recipient at index %d failed to decode: %v", i, err), http.StatusBadRequest)
+			return
+		}
+		if len(feeRecipientBytes) != fieldparams.FeeRecipientLength {
+			http2.HandleError(w, fmt.Sprintf("Invalid fee recipient address %s at index %d", r.FeeRecipient, i), http.StatusBadRequest)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/handlers.go:610]: This message should clearly state that it's the length that is wrong.

File: beacon-chain/rpc/eth/validator/handlers.go [hunk 4]
```diff
@@ -571,3 +577,62 @@ func (s *Server) RegisterValidator(w http.ResponseWriter, r *http.Request) {
...
+	var validatorIndices []primitives.ValidatorIndex
+	// filter for found fee recipients
+	for i, r := range jsonFeeRecipients {
+		validatorIndex, valid := shared.ValidateUint(w, "Validator Index", r.ValidatorIndex)
+		if !valid {
+			return
+		}
+		feeRecipientBytes, err := hexutil.Decode(r.FeeRecipient)
+		if err != nil {
+			http2.HandleError(w, fmt.Sprintf("fee recipient at index %d failed to decode: %v", i, err), http.StatusBadRequest)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/handlers.go:606]:
```suggestion
			http2.HandleError(w, fmt.Sprintf("Fee recipient at index %d failed to decode: %v", i, err), http.StatusBadRequest)
```

File: beacon-chain/rpc/eth/validator/handlers.go [hunk 5]
```diff
@@ -571,3 +577,62 @@ func (s *Server) RegisterValidator(w http.ResponseWriter, r *http.Request) {
...
+	if len(validatorIndices) == 0 {
+		return
+	}
+	if err := s.BeaconDB.SaveFeeRecipientsByValidatorIDs(ctx, validatorIndices, feeRecipients); err != nil {
+		http2.HandleError(w, fmt.Sprintf("Could not save fee recipients: %v", err), http.StatusInternalServerError)
+		return
+	}
+	log.WithFields(log.Fields{
+		"validatorIndices": validatorIndices,
+	}).Info("Updated fee recipient addresses for validator indices")
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/handlers.go:637]:
```suggestion
	}).Info("Updated fee recipient addresses")
```

File: beacon-chain/rpc/eth/shared/structs.go [hunk 6]
```diff
@@ -86,6 +86,11 @@ type SignedValidatorRegistration struct {
 	Signature string                 `json:"signature" validate:"required,hexadecimal"`
 }
 
+type FeeRecipient struct {
+	ValidatorIndex string `json:"validator_index"`
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/shared/structs.go:90]: doesn't have validation here as we are moving away from it

**Pull request merged**